spotbugs/src/main/java/edu/umd/cs/findbugs/ba/LongRangeSet.java

```diff
@@ -0,0 +1,321 @@
+package edu.umd.cs.findbugs.ba;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.concurrent.ConcurrentSkipListMap;
+import java.util.Map.Entry;
+
+public class LongRangeSet implements Iterable<LongRangeSet> {
+    private static final Map<String, TypeLongRange> typeRanges;
+
+    static {
+        typeRanges = new HashMap<>();
+        typeRanges.put("Z", new TypeLongRange(0, 1, "Z"));
+        typeRanges.put("B", new TypeLongRange(Byte.MIN_VALUE, Byte.MAX_VALUE, "B"));
+        typeRanges.put("S", new TypeLongRange(Short.MIN_VALUE, Short.MAX_VALUE, "S"));
+        typeRanges.put("I", new TypeLongRange(Integer.MIN_VALUE, Integer.MAX_VALUE, "I"));
+        typeRanges.put("J", new TypeLongRange(Long.MIN_VALUE, Long.MAX_VALUE, "J"));
+        typeRanges.put("C", new TypeLongRange(Character.MIN_VALUE, Character.MAX_VALUE, "C"));
+    }
+
+    public static boolean isSignatureSupported(String signature) {
+        return typeRanges.containsKey(signature);
+    }
+
+    private SortedMap<Long, Long> map = new ConcurrentSkipListMap<>();
```

Comment:
No, this was a misunderstanding. I got several ConcurrentModificationExceptions but they were because I tried to merge the same instance into itself. A check was missing to prevent this.